src/overlays/actors/ovl_Dm_Zl/z_dm_zl.c

```diff
@@ -30,27 +30,246 @@ const ActorInit Dm_Zl_InitVars = {
     (ActorFunc)DmZl_Draw,
 };
 
-#endif
+static AnimationInfo sAnimations[7] = {
+    { &gDmZl4FacingAwayHandsOverEmblemLoop, 1.0f, 0.0f, -1.0f, ANIMMODE_LOOP, -10.0f },
+    { &gDmZl4TurningAround2Anim, 1.0f, 0.0f, -1.0f, ANIMMODE_ONCE, -10.0f },
+    { &gDmZl4HandsOverEmblemLoopAnim, 1.0f, 0.0f, -1.0f, ANIMMODE_LOOP, -10.0f },
+    { &gDmZl4GivingItemStartAnim, 1.0f, 0.0f, -1.0f, ANIMMODE_ONCE, -10.0f },
+    { &gDmZl4GivingItemLoopAnim, 1.0f, 0.0f, -1.0f, ANIMMODE_LOOP, -10.0f },
+    { &gDmZl4RaisingOcarinaToPlayAnim, 1.0f, 0.0f, -1.0f, ANIMMODE_ONCE, -10.0f },
+    { &gDmZl4PlayingOcarinaAnim, 1.0f, 0.0f, -1.0f, ANIMMODE_LOOP, -10.0f },
+};
+
+typedef enum {
+    ZELDA_ANIM_FACING_AWAY,
+    ZELDA_ANIM_TURNING_TOWARD_PLAYER,
+    ZELDA_ANIM_FACING_PLAYER,
+    ZELDA_ANIM_GIVING_OCARINA_START,
+    ZELDA_ANIM_GIVING_OCARINA,
+    ZELDA_ANIM_PLAYING_OCARINA_START,
+    ZELDA_ANIM_PLAYING_OCARINA,
+    ZELDA_ANIM_MAX,
+} DmZlAnimations;
+
+static TexturePtr sMouthTextures[4] = {
+    gZl4MouthNeutralTex,
+    gZl4MouthOpenSmilingTex,
+    gZl4MouthFrowningTex,
+    gZl4MouthOpenTex,
+};
+
+typedef enum {
+    /* 0 */ ZELDA_MOUTH_NEUTRAL,
+    /* 1 */ ZELDA_MOUTH_SMILING,
+    /* 2 */ ZELDA_MOUTH_FROWNING,
+    /* 3 */ ZELDA_MOUTH_OPEN,
+} DmZlMouthTextures;
+
+static TexturePtr sEyeTextures[] = {
+    gDmZl4EyeOpenNormalTex, gDmZl4EyeBlinkingTex,        gDmZl4EyeClosedTex,           gDmZl4EyeWideTex,
+    gDmZl4EyeHappyTex,      gDmZl4EyeOpenLookingLeftTex, gDmZl4EyeOpenLookingRightTex,
+};
+
+typedef enum {
+    /* 0 */ ZELDA_EYE_OPEN_NORMAL,
+    /* 1 */ ZELDA_EYE_BLINKING,
+    /* 2 */ ZELDA_EYE_CLOSED,
+    /* 3 */ ZELDA_EYE_WIDE,
+    /* 4 */ ZELDA_EYE_HAPPY,
+    /* 5 */ ZELDA_EYE_OPEN_LOOKING_LEFT,
+    /* 6 */ ZELDA_EYE_OPEN_LOOKING_RIGHT,
+} DmZlEyeTextures;
+
+void DmZl_ChangeAnimation(SkelAnime* skelAnime, AnimationInfo animation[], u16 index) {
```

Comment:
I remain unconvinced that these array arguments are actually helpful. The argument should be an `animationInfo` or `animInfo` anyway, since the actual animation is just one of the fields.

Reply:
That doesn't explain what the third value is doing, if its getting an AnimationInfo passed, what does `<AnimationInfo> + 1` evaluate to? This C makes no sense either way. 

Right now, I have to assume the function was designed so that there would be double animations in the array, where there is an offset of one for a different age or different... something, some modifier that would get evaluated last second, but it was unused by this actor because its a 3 minute cutscene actor cutdown from a bigger original design.